airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcDestinationHandler.kt

```diff
@@ -41,19 +41,28 @@ import org.slf4j.LoggerFactory
 
 @Slf4j
 abstract class JdbcDestinationHandler<DestinationState>(
-    protected val databaseName: String,
+    // This feels hacky. MySQL doesn't have a notion of "schema".
+    // In this case, the JDBC interface translates the schema parameter to mysql database.
+    // This allows us to pass e.g. stream namespace into metadata.get(schema=?) for mysql.
+    // So it's misleading, but mysql will use a null databaseName here to avoid problemss.
+    protected val databaseName: String?,
```

Comment:
seems super hacky. Especially since the comment mentions schema, but the `database` parameter is nullable...

Reply:
I think there's (yet another) an abstraction issue here. I think your suggestions make things clearer though, so let's go with those for now